### PR TITLE
Add envs ipfshash

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

This PR does mainly two things:
* Uses a different AppData code per environemnt
* Makes further differentiation introducing new environments: `Local`, `PR`

image

image

Note that it also adds a new metadata `environment` in the App Data JSON file:
```json
{
  "version": "0.1.0",
  "appCode": "CowSwap",
  "metadata": {
    "environment": "barn"
  }
}
```

Additionally, it renames `prestaging` environment to `barn`, just to be more clear.

The environment is exposed also as a global var (window.environment), to make it easier to read. 
This PR also affects to sentry, now we have more environments, so we have further details where the errors come from.

image

# To Test

1. Open the app link, and check in the console that the environment is `pr`. 
2. Sign an order, and verify the `appData` you sign matches the `pr` one.

Test also locally, and check in the console that the environment is `local`. 
1. Run app locally
2. Sign an order, and verify the `appData` you sign matches the `local` one.

# Background
- We want to segment the volume also by environment if needed.
- Also nice to differentiate in sentry, so we know if the erros are from dev, pr or local

### CODE REVIEW

**Comment by josojo** [comment]
Looks good. Makes sense

**Review by W3stside — APPROVED** [review]
nice

File: src/custom/utils/analytics.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { isDev, isStaging, isProd, isEns, isPreStaging } from './environments'
+import { isLocal, isDev, isPr, isStaging, isProd, isEns, isBarn } from './environments'
 
 function getAnalyticsId(): string | undefined {
-  if (isDev || isPreStaging) {
+  if (isLocal || isDev || isPr || isBarn) {
```
Comment (alfetopito) [inline src/custom/utils/analytics.ts:4]: Maybe barn should use the same analytics as staging? Or even it's own.

File: README.md [hunk 2]
```diff
@@ -78,26 +78,29 @@ Make a copy of `.env` named `.env.local`, this will allow you to set your own co
...
-id is included in all signed transaction, although the Gnosis Protocol is not using this information for now, it
-could be used for implementing incentive programs.
+To set your own `AppData`, change `REACT_APP_DOMAIN_REGEX_<environment>` environment variable. Ask for yours at [chat.cowswap.exchange/](https://chat.cowswap.exchange/). For more details, check out the environment file (<.env>)
 
-To set your own, change `REACT_APP_DATA_HASH` environment variable. Ask for yours at [chat.cowswap.exchange/](https://chat.cowswap.exchange/)
+REACT_APP_DOMAIN_REGEX_LOCAL="^(:?localhost:\d{2,5}|(?:127|192)(?:\.[0-9]{1,3}){3})"
+REACT_APP_DOMAIN_REGEX_PR="^pr\d+--gpswapui\.review"
+REACT_APP_DOMAIN_REGEX_DEV="^cowswap\.dev"
+REACT_APP_DOMAIN_REGEX_STAGING="^cowswap\.staging"
+REACT_APP_DOMAIN_REGEX_PROD="^cowswap\.exchange$"
```
Comment (alfetopito) [inline README.md:101]: Why only prod regex ends with `$`?

File: .env [hunk 3]
```diff
@@ -1,5 +1,12 @@
 # App Id, get yours at https://chat.cowswap.exchange
-REACT_APP_DATA_HASH=0x816BBB5787AA2B72E2DE870A4E935A49A97305DE32E637B8126D07BEE10D03AB
+REACT_APP_DATA_HASH_PRODUCTION=0x487B02C558D729ABAF3ECF17881A4181E5BC2446429A0995142297E897B6EB37
+REACT_APP_DATA_HASH_ENS=0xBAB87DF726E41C0941786CA710194982D753FBC140C6DC9951AC3450D8917699
+REACT_APP_DATA_HASH_BARN=0xE4D1AB10F2C9FFE7BDD23C315B03F18CFF90888D6B2BB5022BACD46AB9CDDF24
+REACT_APP_DATA_HASH_STAGING=0xA076A100C2535DC6047C4C9940AE647D7DEAAC1729745117D19D4A63BC2F4D30
+REACT_APP_DATA_HASH_PR=0x0BCAECDB9A1FDB3B207A7593EBF703836AD591D4E5E75DFDBF65E7B328F209CD
+REACT_APP_DATA_HASH_DEVELOPMENT=0x828569F802B7F8957F76996BDD875674821E41A688541A9E9EC97D5E897D44A7
+REACT_APP_DATA_HASH_LOCAL=0xA5DAE7A114F1BD6BB9B3FF976150380A95CB18856212DB555C25EF9D7801E9A4
```
Comment (alfetopito) [inline .env:8]: Oh boy, that's a lot of environments

**Review by alfetopito — APPROVED** [review]

**Comment by alfetopito** [comment]
FYI @ramirotw who worked on this before.
I'll update whatever changes are necessary when defining the IPFS task, but just so you know

File: src/custom/utils/analytics.ts (hunk 1 above)
Comment (anxolin) [reply src/custom/utils/analytics.ts:4]: okok

File: README.md (hunk 2 above)
Comment (anxolin) [reply README.md:101]: cause $ is end of the pattern. The other are just some prefixes.

well, u are right about barn and ens. Will add

File: src/custom/utils/environments.ts [hunk 4]
```diff
@@ -1,41 +1,53 @@
...
     return 'development'
+  } else if (isLocal) {
+    return 'local'
   } else {
     return undefined
   }
 })()
 
-export { isDev, isStaging, isProd, isEns, isPreStaging }
+registerOnWindow({ environment: environmentName })
```
[the remark this answers is not in the document]
Comment (W3stside) [reply src/custom/utils/environments.ts:51]: no? @anxolin

[the remark this answers is not in the document]
Comment (anxolin) [reply src/custom/utils/environments.ts:51]: could be, would be weird. We can also modify `registerOnWindow` to make sure it never overrides something.

`window.xxxx = window.xxxx || yourStuff`

File: README.md (hunk 2 above)
Comment (alfetopito) [reply README.md:101]: Actually, for ENS it'll break with cowswap.eth.link, no?

Comment (anxolin) [reply README.md:101]: Handled here: https://github.com/gnosis/cowswap/pull/1292

**Pull request merged**